### ISSUE TITLE
The option in the favorite color filter does not receive focus. In order for a switch user to select a color, they must simulate a mouse click using the move-and-click feature.

### ISSUE THREAD

**Issue by chlane** [vmware/clarity#4611]
[Clarity Version]
3.1.1

[Test URL]
https://clarity.design/documentation/datagrid/custom-filtering	

[WCAG 2.1 Guideline]
2.1.1 Keyboard 
(Level A)
https://www.w3.org/TR/WCAG21/#keyboard

[OS]
MacOS 10.15.3, Windows 10

[User Agents/Assistive Technologies]
Safari 13.0.5, Firefox 75.0, Switch

[Component]
Datagrid

[Description]
The option in the favorite color filter does not receive focus. In order for a switch user to select a color, they must simulate a mouse click using the move-and-click feature.	

[User Impact]
Filtering by color is difficult for switch users because they have to simulate a mouse click instead of being able to tab to the color that they want to choose. The color icons are small, making the target of the simulated mouse click difficult for the user.

[Steps to Reproduce]
See the following instructions for switch usage then follow the steps below.

Windows (see Scan through keys)
https://support.microsoft.com/en-us/help/10762/windows-use-on-screen-keyboard

Mac
https://support.apple.com/en-us/HT202865

1. Using a switch, navigate to and activate the filter icon on the favorite color column using the Tab key. 
2. Try navigating to a color using the Tab key. 
For MacOS only: 
3. Navigate to and activate Pointer --&gt; Mouse Click. 
4. Select the vertical area of the click, and then the vertical coordinate of the color.
5. Select the horizontal area of the click, and then the horizontal coordinate of the color.

[Recommendation]
Span elements with class=&quot;color-square color-selectable ng-star-inserted&quot; must be converted into button elements. This will ensure these components receive keyboard focus. In addition the buttons must provide an indication of state (pressed or not - this will make the &#039;color-selected&#039; class extraneous), along with an accessible name.	&quot;&lt;button class=&quot;&quot;color-square color-selectable ng-star-inserted&quot;&quot; aria-pressed=&quot;&quot;true&quot;&quot; aria-label=&quot;&quot;Magenta&quot;&quot;&gt;
&lt;/button&gt;&quot;

[Reporter Email] 
daman@damanw.com

**Comment by gnomeontherun** [comment]
Hi there, this is an automated message.

In an effort to dedupe and track accessibility reports consistently, this issue is being closed and tracked internally with our Accessibility team. It will still be triaged and addressed as part of ongoing work.
